src/Symfony/Component/Validator/Constraints/TypeValidator.php

```diff
@@ -33,22 +33,25 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        $type = strtolower($constraint->type);
-        $type = 'boolean' == $type ? 'bool' : $constraint->type;
-        $isFunction = 'is_'.$type;
-        $ctypeFunction = 'ctype_'.$type;
-
-        if (\function_exists($isFunction) && $isFunction($value)) {
-            return;
-        } elseif (\function_exists($ctypeFunction) && $ctypeFunction($value)) {
-            return;
-        } elseif ($value instanceof $constraint->type) {
-            return;
+        $types = (array) $constraint->type;
+
+        foreach ($types as $type) {
+            $type = strtolower($type);
+            $type = 'boolean' == $type ? 'bool' : $type;
+            $isFunction = 'is_'.$type;
+            $ctypeFunction = 'ctype_'.$type;
+            if (\function_exists($isFunction) && $isFunction($value)) {
+                return;
```

Comment:
not sure all the "return" are correct: don't we want to check all values in the list?
please add a corresponding test case.

Reply:
We are iterating the types here. As soon as the value matches one of the accepted types we need to return.